src/Core/src/Exception/Http/HttpExceptionTrait.php

```diff
@@ -66,41 +68,28 @@ public function __construct(ResponseInterface $response)
             }
         }
 
-        // Try to guess a better error message using common API error formats
-        // The MIME type isn't explicitly checked because some formats inherit from others
-        // Ex: JSON:API follows RFC 7807 semantics, Hydra can be used in any JSON-LD-compatible format
-        if ($isJson && $body = json_decode($response->getContent(false), true)) {
-            if (isset($body['__type'])) {
-                $parts = explode('#', $body['__type'], 2);
-                $this->awsCode = $parts[1] ?? $parts[0];
-                $message .= "\n\n" . $body['__type'] . "\n\n";
-            }
-
-            if (isset($body['hydra:title']) || isset($body['hydra:description'])) {
-                // see http://www.hydra-cg.com/spec/latest/core/#description-of-http-status-codes-and-errors
-                $separator = isset($body['hydra:title'], $body['hydra:description']) ? "\n\n" : '';
-                $message = ($body['hydra:title'] ?? '') . $separator . ($body['hydra:description'] ?? '');
-            } elseif (isset($body['title']) || isset($body['detail'])) {
-                // see RFC 7807 and https://jsonapi.org/format/#error-objects
-                $separator = isset($body['title'], $body['detail']) ? "\n\n" : '';
-                $message = ($body['title'] ?? '') . $separator . ($body['detail'] ?? '');
-            } elseif (isset($body['message'])) {
-                $this->awsMessage = $body['message'];
-                $message .= "\n\n" . $body['message'] . "\n\n";
-            } elseif (isset($body['Message'])) {
-                $this->awsMessage = $body['Message'];
-                $message .= "\n\n" . $body['Message'] . "\n\n";
-            }
+        $content = $response->getContent(false);
+        if ($isJson && $body = json_decode($content, true)) {
```

Comment:
same here.. matching content type is not enough to assert it's json

Reply:
Are you suggesting we should not do a `json_decode` and see if it succeeds? 

We haven't had any problems with the JSON check before, have we?

Reply:
let me give a concrete example: 
In `InstanceProvider::toArray` I have to duplicate code of `ResponseInterface::toArray` WITHOUT check on content-type of the enpoint was missing, and I couldn't use `$response->toArray()`

```cli
$ curl -i http://169.254.169.254/latest/meta-data/iam/security-credentials/iam.ec2.support.bastion
HTTP/1.0 200 OK
Accept-Ranges: bytes
Content-Length: 1262
Content-Type: text/plain
Date: Sat, 21 Mar 2020 20:27:19 GMT
Last-Modified: Sat, 21 Mar 2020 19:32:19 GMT
Connection: close
Server: EC2ws

{
  "Code" : "Success",
  "LastUpdated" : "2020-03-21T19:31:18Z",
  "Type" : "AWS-HMAC",
  "AccessKeyId" : "ASIA4F4GERHC2QY7S67P",
  "SecretAccessKey" : "XXX",
  "Token" : "XXX",
  "Expiration" : "2020-03-22T01:47:45Z"
}
```

Maybe it's just about the `meta-data` endpoint.. Maybe not.
My suggestion is to:
- when content type is json or xml, we can trust it
- otherwise we should guess the content-type, maybe by trying json_decode, then trying xml decode..

WDYT?

Reply:
I skipped the content type check all together. See https://github.com/async-aws/aws/pull/347